### DISCUSSION TITLE
Add RSS feeds to Discussions

### DISCUSSION THREAD

**Discussion by Jackenmen** [community/community#31] [Discussions]
Discussions would be far more useful for me if I could follow them outside of GitHub easily. RSS feeds provide that and they can also be used by a lot of different automations easily.

I'm thinking of something similar to Discourse so what I want to propose are:
- Feed with all discussions (what you can see by clicking on "Everything" at the left in the Discussions tab)
- Feeds with discussions of a certain type (in case of this repo, it would be a feed of discussions in "Discussions Feedback" and another one for discussions in "Mobile Feedback")
- Feeds for the individual comments in a single discussion (such as all comments in https://github.com/github/feedback/discussions/28)
- Feed with individual comments in all discussions (so it would be the source of the latest comments that has been posted, no matter what discussion they were posted in)

**Comment by becca** [comment]
thank you for taking the time to detail this feedback out for us. if we were to provide webhooks and more granular notification controls would this still not solve your problem? 

where are you hoping to consume all of these notifications and where might be the best place for us to consider serving those notifications to you?

**Reply by Jackenmen** [reply]
Webhooks are only available for repository administrators so that can only help the maintainers unless they also decide to make use of those webhooks to serve this information more publically so that outside contributors can benefit too (and outside contributors are important, repository maintainers aren't the only ones that could want to be active in the community, help other people, etc. and I believe it's important to give those people tools to do so). So while it might help, it depends on how the repository administrators end up using those tools you give them.

As for more granular notification controls, if these were separate from watching the repository (which only has 2 options) and would allow controlling notifications you get by being able to subscribe to whole categories (or all discussions) then it would probably help a lot.

It still limits you to email notifications, and wouldn't allow you to follow discussions that easily from platforms like Slack or Discord (there are integrations that allow you to get updates from RSS feed on those platforms) or just anything else that user could possibly want, but nonetheless, webhooks and **especially** more granular notification controls would still definitely be a huge improvement.

**Reply by kevinmbeaulieu** [reply]
+1 to more granular notification controls- would love to be able to, for example, subscribe to releases and new discussions for a repo without subscribing to _everything_ for that repo

**Reply by HDVinnie** [reply]
would love to see webhook options added

**Reply by Lautenschlager-id** [reply]
Webhook option for Discussions would be great, indeed.

**Reply by tomkerkhove** [reply]
@becca I think webhooks are great but they have a different way of consuming them than RSS feeds (push vs pull).

Every morning I wake up and check my RSS feeds on what's new, while with webhooks you have to actively process them and build tech around it to handle it.

**Reply by Griatch** [reply]
@becca 
> where are you hoping to consume all of these notifications and where might be the best place for us to consider serving those notifications to you?

I support webhook primarily, RSS as a bonus, if possible: The main day-to-day support/community interaction for our [evennia](https://github.com/evennia/evennia) project happens through IRC/Discord. Our current google-groups mailing list (hopefully to be replaced with Discussions) today acts as the more slow-moving persistent-form support/discussion/announcements for those that can't hang around in chat. All post-events on the ML are echoed into the chat channels so everyone can keep up. We use github webhooks in the same way - reporting new commits/PRs etc into the support chat. Discussion-webhooks would thus be necessary for us to switch to Discussions and integrate them into our existing community structure, replacing our current mailing list.

... RSS feeds would be great in addition of course, but more from the perspective of the individual than of the maintainer. In lieu of webhooks, we could also poll RSS feeds and echo that to our bots, but I'd prefer a proper webhook for this. :)

**Reply by jdrch** [reply]
> thank you for taking the time to detail this feedback out for us. if we were to provide webhooks and more granular notification controls would this still not solve your problem?

Not for me. I'm hoping to make GitHub Discussions my new blogging platform, since the repo it's on for me is what I blog about the most. That works best if Discussions has a simple RSS feed for all new discussion threads.

**Reply by Mikaela** [reply]
I would like to get a simple RSS feed out on my activity in discussions like there is https://github.com/Mikaela.atom and feed them into an IRC bot to have a unified view on my activity somewhere and be able to look back and see that I was contributing to discussions, while currently discussions activity can only be seen within the discussions themselves.

**Reply by alvinashcraft** [reply]
Some of the .NET teams at Microsoft are starting to use discussions as a replacement for blog posts when announcing releases. Here are two examples:

- [.NET MAUI Announcements](https://github.com/dotnet/maui/discussions/categories/announcements)
- [.NET News](https://github.com/dotnet/core/discussions/categories/news)

Some people will probably be happy to have less noise in the RSS feed for the .NET blog, but I'll be missing those posts in my RSS reader ([Newsblur](https://www.newsblur.com/)). Adding RSS feeds to discussion categories would be extremely helpful to track release announcements and other news for projects.

**Comment by eeholmes** [comment]
Adding to dashboard (github.com/dashboard) would be great too. Right now the dashboard just shows issues and commits but I can't see any of the recent discussion posts.

**Comment by m-ueberall** [comment]
I'd like to point out that in case you follow large projects, the RSS feed(s) in question should contain references (and/or the full-text, if possible) to _an appropriate number of last postings_ to enable users (and/or cronjobs that don't run every five minutes) to follow/monitor new discussion threads/postings.

This way, tools like [rss2email](https://github.com/wking/rss2email) will allow users to group or combine all/most kinds of updates specific to one or more project(s) using (locally shared) e-mail folders – and, by extension, a unified interface – without fail.

**Comment by mdoar** [comment]
Yes, please. RSS still just works.

**Comment by markcellus** [comment]
This issue has been open for a while. Can anyone from the GitHub team give any update on whether this has been prioritized with the recent Discussion features introduced or whether it is even on the roadmap? We already have RSS feeds for GitHub releases and because the infrastructure for this feature already exists, the implementation of this feature seems quite trivial.

**Comment by placoderm** [comment]
I'm also surprised that this has not been implemented.

Wondering what's holding it up. Is there a better place to raise the issue?

**Comment by sonniesedge** [comment]
Bumping this to see if there is any response from Github folk (@becca ?) . RSS feeds feel like a basic feature for any content publication system.

**Reply by sonniesedge** [reply]
oh sorry @becca you don't work for Github any more. 😂

**Reply by Pythone** [reply]
agree

**Comment by tyrosinase** [comment]
Gitlab has RSS feeds...

**Comment by magnus919** [comment]
I think if I were to put a finer point on it... any time I build a search query that gives me results I'm interested in, I'd love very much to have an RSS link to those search results. So for example I could follow all Discussions on a project where Categories match "Announcements" to stay plugged into the community updates from the project maintainers.

Or I might want to follow all issues that get labeled as "feature request".

This is such a good idea for so many reasons. Webhooks are for machines. RSS feeds are for people who want to follow interesting things.

**Comment by jordan-brough** [comment]
Another use case: One third-party framework I use has a "Releases" Category in Discussions (they don't use the "Releases" GitHub tab) and I'd like to get Slack notifications when new discussions are posted there. If I could get an RSS link that returns those results I could set up Slack to notify me based on the RSS feed.

**Comment by aborruso** [comment]
Please add the discussion RSS feed, it could be useful for automatic triggering on any web space, because RSS is a standard, and a lot of tools are able to use in input RSS to post about its update in a lot of webspace output

**Comment by jiacai2050** [comment]
Almost two years pass, still no progress on this? @github-staff

**Comment by sonniesedge** [comment]
RSS feeds aren't compatible with walled gardens...

**Reply by Pythone** [reply]
We need to work on an open source alternative to discussions then.

**Reply by lersi** [reply]
GitHub is shit, use gitlab instead

**Reply by tyrosinase** [reply]
Not terribly useful when the projects you want to follow are on GitHub and you have no control over it.

I mean: why didn't *you* post your response on Gitlab?

**Comment by hmans** [comment]
Webhooks for Discussions [appear to be in beta at the moment](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#discussion). They will at lease enable someone to build a service that

- receives these webhooks
- mirrors the data (if this isn't against GitHub's TOS?)
- serves RSS feeds containing the data.

The feeds wouldn't be advertised from here, but at least it's a way to make them _available_.

All in all I'd prefer if GitHub just added their own feeds though.

**Reply by jdrch** [reply]
Some progress is better than none :)

**Comment by markcellus** [comment]
Hey everyone, I've made RSS feeds for Github Discussions available on Open RSS, a non-profit project I developed to provide RSS feeds for websites that don't have them. 

You can get a Discussion RSS feed by navigating to any discussions dashboard page and typing in `openrss.org/` right before the `github.com` part in the browser URL bar. e.g. https://openrss.org/github.com/octokit/octokit.rb/discussions.  Works the same for issues and pull requests as well. Here's a [guide](https://openrss.org/feeds/github) on how to use them.

**Reply by aborruso** [reply]
> For example, the RSS feed for the discussions at https://github.com/octokit/octokit.rb/discussions are located at https://openrss.org/github.com/octokit/octokit.rb/discussions

This is really great, thank you @markcellus 

Where do I find the list of all websites supported?

**Reply by czottmann** [reply]
@markcellus As someone who made some decent pocket money in the '00s by building RSS feeds for sites and people that didn't have them — thanks for doing that. Keep up the good work, RSS is still very much needed, and you're a legend for fighting the good fight here. 🖖🏼

**Reply by tabacitu** [reply]
@markcellus how often does OpenRSS refresh a Discussion feed?

I have a Github repo that's had new discussions in the past 4 days, but they don't show up in the RSS feed at all - https://openrss.org/github.com/Laravel-Backpack/community-forum/issues

**Reply by amarouane-ABDELHAK** [reply]
@markcellus the discussion feed does not update once generated. The github RSS feed fir the discussion is kind of useless since it doesn't target one discussion and get the comments :sad_face:

**Comment by magnus919** [comment]
Please don't hijack this issue to talk about another product. Thanks.

**Reply by dpolivaev** [reply]
What do you mean by "another product"? openrss?
They have provided at least a workaround. If the feature request becomes implemented directly at GitHub, the workaround would not be needed. But for the moment it should be praised here as the only available solution.

**Reply by tyrosinase** [reply]
Mentioning it is a good thing, but talking about its features etc. keeps notifying people who are following this in probably-vain hope for a native fix.

(Now, if we were following via RSS it wouldn't be so bad, but instead we get new-mail pings...)

**Comment by jayceeday** [comment]
Hey everyone!  Jaycee here, Designer on Discussions at GitHub. 

Admittedly, this took us a while to get around to, but we did it. 

RSS feeds are now available for Discussions 🎉

Here you go: 

https://github.com/OWNER/REPO/discussions.atom
ex. https://github.com/community/community/discussions.atom

https://github.com/OWNER/REPO/discussions/categories/CATEGORY.atom
ex. https://github.com/community/community/discussions/categories/announcements.atom

**Reply by hmans** [reply]
**Thank you very much**, this is a very welcome new addition!

As another user already hinted at, adding the discovery tags to the HTML versions would be very much appreciated. This would simply be a matter of adding headers like these to the relevant pages:

```html
<link
  rel="alternate"
  type="application/atom+xml"
  title="Community Announcements" 
  href="https://github.com/community/community/discussions/categories/announcements.atom" />
```

It would allow users to paste URLs to the web version of discussions into feed readers and have those pick up the feeds, and also allow feed-discovery browser plugins to work.

But that would really be just the icing on the cake. Thank you very much for giving us feeds!

**Reply by cwalther** [reply]
Yay, thanks!

Am I understanding correctly that the feeds only include the first post of each discussion? That greatly reduces their usefulness. To actually be able to follow discussions from my feed reader I would need a feed that includes all posts.

(Although, I’ve since discovered that my needs for a standardized local offline archive are adequately met by e-mail notifications, so I’m no longer as reliant on Atom feeds as I thought I would be.)

**Reply by HansRobo** [reply]
Thank you，It's a great feature!
I have already used it usefully for some GitHub Discussions.

However, at present, it seems that the GitHub discussion of rrganization is not covered, so I would be happy if you could consider additional support.

**Reply by markcellus** [reply]
@HansRobo Until org-wide discussions are added to Github natively, I've added RSS feeds for org-wide discussions through openrss at 

`https://openrss.org/github.com/orgs/ORG/discussions`
ex. https://openrss.org/github.com/orgs/community/discussions

**Reply by tyrosinase** [reply]
@markcellus Hi yeah you're still pinging everybody who is following this whenever you push your service; once was enough.

**Reply by dpolivaev** [reply]
@tyrosinase I think that markcellus is not "pushing services" for his own interest. He is trying to help people by sharing with them free solutions. I wish everyone to appreciate his honorable contributions.
Kind regards.

**Reply by tyrosinase** [reply]
Except it keeps pinging everyone with an email notification every single time, which of course would not happen if we could get these comments via RSS, but we can't.

**Reply by dpolivaev** [reply]
If his solution does not allow you to get comment updates using RSS yet, you could ask him if he could improve his solution.
I think it is more constructive than blaming a good man for his good intentions.

**Reply by hmans** [reply]
> Except it keeps pinging everyone with an email notification every single time, which of course would not happen if we could get these comments via RSS, but we can't.

I've received more notifications about your angry replies that nobody asked for than for @markcellus legitimately useful comments. I would recommend you unsubscribe from this discussion and maybe check manually once a year to see if your favorite feature finally got implemented. Thank you.

**Reply by jazzsequence** [reply]
I don't see this working in _team_ discussions (e.g. `github.com/orgs/<org>/teams/<team>/discussions`) as opposed to _repository_ discussions. Is that in scope for the RSS (Atom) feature?

**Reply by sunzoje** [reply]
How does this work with private repositories?

**Reply by LindseyB** [reply]
@jazzsequence You are correct, this does not work for team discussions.

**Reply by cwalther** [reply]
> https://github.com/OWNER/REPO/discussions.atom ex. https://github.com/community/community/discussions.atom

This seems to have stopped working a few days ago. https://github.com/community/community/discussions.atom now redirects to https://github.com/orgs/community/discussions, which is HTML, not Atom, and https://github.com/orgs/community/discussions.atom (tried on a hunch) returns “406 Not Acceptable”. Where have the feeds gone?

**Reply by LindseyB** [reply]
This is a bug. I've reported.

**Reply by LindseyB** [reply]
This is fixed now. Thanks again for the report.

**Reply by cwalther** [reply]
Confirmed, the feeds work again. Thanks!

**Reply by edu18RR** [reply]
La propuesta es implementar una función de "Búsqueda y Filtrado Avanzado" en la sección de discusiones de GitHub. Esta función permitiría a los usuarios buscar y filtrar discusiones de manera personalizada según sus intereses y preferencias.

Beneficios clave:

1. Acceso rápido a discusiones relevantes: Los usuarios podrán utilizar palabras clave y filtros específicos para encontrar rápidamente las discusiones que se ajusten a sus intereses. Esto les ahorrará tiempo y esfuerzo al evitar la necesidad de revisar manualmente cada repositorio.

2. Mantenerse al día con temas de interés: La búsqueda y el filtrado avanzado permitirán a los usuarios mantenerse actualizados con los temas que les interesan sin perderse información importante en medio de múltiples discusiones.

3. Mejor experiencia de usuario: Al facilitar la búsqueda de discusiones, GitHub mejorará la experiencia del usuario, lo que podría aumentar la participación y la colaboración dentro de la comunidad.

**Características propuestas:**

1. **Búsqueda por palabras clave**: Los usuarios podrán ingresar palabras clave relacionadas con el tema que están buscando, y el sistema mostrará las discusiones relevantes.

2. **Filtros personalizados**: Los usuarios podrán aplicar filtros según la fecha de publicación, etiquetas, estado de la discusión (abierto/cerrado), entre otros.

3. **Guardar búsquedas**: Los usuarios tendrán la opción de guardar búsquedas frecuentes para acceder rápidamente a las discusiones relevantes en el futuro.

4. Notificaciones de búsqueda: Los usuarios pueden optar por recibir notificaciones cuando se publiquen nuevas discusiones que coincidan con sus búsquedas guardadas.

5. Búsquedas populares: Proporcionar una lista de búsquedas populares y tendencias en la comunidad para ayudar a los usuarios a descubrir nuevos temas de interés.

Al implementar una función de "Búsqueda y Filtrado Avanzado" en las discusiones de GitHub, se facilitará el acceso a la información relevante y se enriquecerá la experiencia de los usuarios en la plataforma. Esto promoverá una mayor participación y colaboración dentro de la comunidad, asegurando que los usuarios se mantengan actualizados con los temas que más les interesan.

**Reply by bogorad** [reply]
How about this as a single feed? Or better yet, any filter as a feed!

https://github.com/notifications?query=reason%3Aparticipating

**Reply by segln** [reply]
<https://github.com/orgs/community/discussions/31.atom> returns `406 Not Acceptable`.

Are Atom feeds for specific discussion threads unimplemented, or is this a bug?

**Comment by tyrosinase** [comment]
A good first step, but a quick submission of https://github.com/community/community/ to my RSS reader says it can't auto-detect the feed. I can manually put it in, but discoverability is important.

Until then, what's the URL for the version with comments?

**Reply by LindseyB** [reply]
There is no URL version for the comments at the comment right now you can use https://github.com/community/community/discussions.atom, but would have to follow the links to get to the comments.

**Comment by markcellus** [comment]


**Reply by Pythone** [reply]
> Hey, I appreciate everyone here and your overall support for Github feeds provided by Open RSS! heart But like others have said, we should probably keep non-Github related discussion out of this thread. I'm happy to discuss any questions, concerns or feedback related to Open RSS feeds by just emailing me at mark[at]openrss[dot]org.

GH discussions is a part of GH. This is not just about the feeds, this is about if they are properly implemented.

**Comment by Sandesh9892** [comment]
When I try to verify for the student on Git Hub for accessing the student package in copilot, it always says that I am not verified even when I tried every document, card, or student card that the university gave me.
What do you think I should do?
